src/app/shared/modules/sai-editor/sai-comment-generator.service.ts

```diff
@@ -475,6 +489,55 @@ export class SaiCommentGeneratorService {
           break;
       }
     }
+	
+	if (actionLine.indexOf('_waypointStartActionParamThree_') > -1) {
+      switch (Number(smartScript.action_param1)) {
+        case 0:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', 'Waypoint ');
+          break;
+        case 1:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', 'Patrol ');
+          break;
+        default:
+          actionLine = actionLine.replace('_waypointStartActionParamThree_', '[Unknown Value]');
+          break;
+      }
```

Comment:
The only part that changes can be isolated to reduce the redundancy.
```suggestion
      let waypointReplace: string;
      switch (Number(smartScript.action_param1)) {
        case 0:
          waypointReplace = 'Waypoint ';
          break;
        case 1:
          waypointReplace = 'Patrol ';
          break;
        default:
          waypointReplace = '[Unknown Value]';
          break;
      }

      actionLine = actionLine.replace('_waypointStartActionParamThree_', waypointReplace);
```